src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,339 @@
+                    addTodo(task, false, true);
+                    break;
+                case COMMAND_DEADLINE:
+                    addDeadline(task, false, true);
+                    break;
+                case COMMAND_EVENT:
+                    addEvent(task, false, true);
+                    break;
+                case COMMAND_DELETE:
+                    deleteTask(task);
+                    break;
+                default:
+                    throw new DukeException(ERROR_UNRECOGNISED_INPUT);
+                }
+            } catch (NumberFormatException e) {
+                printError(ERROR_INVALID_INPUT);
+            } catch (IndexOutOfBoundsException e) {
+                printError("Number is out of range!");
+            } catch (DukeException e) {
+                printError(e.errorMessage);
+            }
+
+            try {
+                writeFile(file, newDukeList());
+            } catch (IOException e) {
+                printError(e.getMessage() + "\n Please create a file named "
+                        + file + " !");
+            }
+
+            in = new Scanner(System.in);
+            line = in.nextLine();
+            }
+        }
+
+    private static void readFile(String filePath) throws FileNotFoundException, DukeException {
+        File file = new File(filePath);
+        Scanner scan = new Scanner(file);
+
+        while(scan.hasNext()) {
+            extractData(scan.nextLine());
+        }
+    }
+
+    private static void extractData(String line) throws DukeException {
+        String[] args = line.split("\\|");
+        String taskType = args[0].trim();
+        boolean isDone = Integer.parseInt(args[1].trim()) == 1;
+        String description = args[2].trim();
+
+        switch (taskType) {
+        case "T":
+            addTodo(description, isDone, false);
+            break;
+        case "D":
+            description = description + " /by " + args[3].trim();
+            addDeadline(description, isDone, false);
+            break;
+        case "E":
+            description = description + " /at " + args[3].trim();
+            addEvent(description, isDone, false);
+            break;
+        default:
+            printError("Cannot read task!");
+            break;
+        }
+    }
+
+    public static void main(String[] args) {
+        String directoryName = "data";
+        String filePath = "data/duke.txt";
+        File file = new File(filePath);
+        File data = new File(directoryName);
+
+        try {
+            data.mkdir();
+            file.createNewFile();
+            readFile(filePath);
+        } catch (FileNotFoundException e) {
+            printError("File not found");
+        } catch (IOException e) {
+            printError(e.getMessage());
+        } catch (DukeException e) {
+            printError(e.errorMessage);
+        }
+
+        System.out.println(MESSAGE_GREET_SIGN);
+        handleTasks();
+        System.out.println(MESSAGE_BYE_SIGN);
+    }
+}
```

Comment:
The duke.java file is a bit too long. How about reducing its length by using separate classes for file managing and task managing?